sdv/single_table/base.py

```diff
@@ -56,18 +56,22 @@ def _update_default_transformers(self):
             for sdtype, transformer in self._model_sdtype_transformers.items():
                 self._data_processor._update_transformers_by_sdtypes(sdtype, transformer)
 
+    def _initialize_synthesizer(self):
+        self._data_processor = DataProcessor(
+            metadata=self.metadata,
+            enforce_rounding=self.enforce_rounding,
+            enforce_min_max_values=self.enforce_min_max_values
+        )
+        self._update_default_transformers()
+        self._fitted = False
+        self._random_state_set = False
+
```

Comment:
When fitting was called over again it wasn't resetting properly the rest of the items. That is why I decided to create this function which should reset the `data_processor` and other states.

Reply:
why does the `DataProcessor` need to be re-initialized? Can't it just refit?

Reply:
Yes I think we can actually re-fit, I can't find the test that I did that was causing this to fail. Maybe I used the wrong data at some point and thats why it didn't work at first glance. But so far it should be working by just refitting, applied the changes.